GET-8390: update PR template, include security in checklist

gemmadlou opened the pull request:
After reviewing an OWASP report [GET-8390](https://kamma.atlassian.net/browse/GET-8390), one way to mitigate against SQL injections is to use prepared database statements. The other way which Gary pointed out was to validate incoming requests to API or web endpoints so that only what is expected can be used in either a URL parameter or a query parameter. Therefore, I've added two new security checkboxes.

I also removed the **detailed description** section in the PR template as the team do a great job of writing good, explicit PRs.

## Related Issue
<!--- Please link to the issue here: -->

https://kamma.atlassian.net/browse/GET-8390

## Screenshots
<!--- Share any sceenshots your project may have that you think might be helpful or QA and documentation. -->

## Known Technical Debt
<!--- Describe any known technical debt issues that have been considered in detail -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes or test areas that should be looked into/prioritised (advisory) by QA. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.


[GET-8390]: https://kamma.atlassian.net/browse/GET-8390?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

gemmadlou committed:
GET-XXXX: update PR template, include security in checklist

jdharding92 reviewed (approved):
Makes sense 👍

othyn commented on .github/pull_request_template.md:
```diff
@@ -34,8 +27,15 @@ https://kamma.atlassian.net/browse/GET-XXXX
+
+### Security and validation
+- [ ] The code has appropriate validation
+- [ ] All raw database queries use prepared statements
```
I like this idea! Not having a security section but two sections on detail does sound silly in retrospect haha

Could we add the following to that list as well?

(the second point should augment the first on the branch version - but GitHub's suggestions feature is a little _wonky_ with the diffs)

```suggestion
- [ ] All raw database queries use prepared statements
- [ ] All external data (e.g. from API's, data sets or form POST) are sanitised,  validated and handled defensively in their entirety
- [ ] Any vulnerability alerts for associated packages are patched (RFC's or otherwise)
- [ ] The base Docker image, if it applies, is up to date with the latest OS image and packages
```

othyn reviewed (changes requested):
Just a few security checklist additions, may think of more!

gemmadlou committed:
GET-8390: add additional security checks to PR checklist

Co-authored-by: Ben <git@othyn.com>

gemmadlou committed:
GET-8390: remove duplicate checkboxes in PR

The new checkboxes are wider in scope and are more explicit as per @othyn's suggestion.

gemmadlou committed:
GET-8390: use subheadings in PR checklist

After some helpful feedback from @othyn, this is to keep things organised and separate.

othyn commented on .github/pull_request_template.md:
```diff
@@ -36,6 +29,14 @@ https://kamma.atlassian.net/browse/GET-XXXX
 - [ ] I have updated the documentation accordingly.
+
+### Security
+
```
Missing the DB line, think its been accidentally deleted between commits

```suggestion

- [ ] All raw database queries use prepared statements
```

gemmadlou replied:
I didn't realise. Thanks @othyn .

vicchi reviewed (changes requested):
@gemmadlou Looks great but can we also add a checklist item for "all tests pass" and somewhere to explain why it's OK for the tests not to pass, such as "requires deployment" or another reasonable excuse?

artisanphil reviewed (approved):
@gemmadlou I like it, but doesn't Laravel handle some of the sanitation out of the box?

gemmadlou commented:
> @gemmadlou I like it, but doesn't Laravel handle some of the sanitation out of the box?

Hi @artisanphil . Good question. Laravel does mostly. You're already aware of `DB::raw` and `DB::statement`, which was my only concern: https://www.stackhawk.com/blog/sql-injection-prevention-laravel/

gemmadlou commented:
> @gemmadlou Looks great but can we also add a checklist item for "all tests pass" and somewhere to explain why it's OK for the tests not to pass, such as "requires deployment" or another reasonable excuse?

Hi @vicchi . That's a good one. I didn't think of that 👍 

I've added a tests section too. Is this sufficient; too much; too little?

<img width="980" alt="image" src="https://user-images.githubusercontent.com/4562670/214596998-61edc355-73de-4a2c-a0ce-50d2b55ffc0b.png">

gemmadlou committed:
GET-8390: Update security section of PR template

Co-authored-by: Ben <git@othyn.com>

gemmadlou committed:
GET-8390: add a tests checklist section to the PR template

This is just to aid and remind PR reviewers and developers about testing.

apepindev reviewed (changes requested):
@gemmadlou Could you also include replacing `GET-` with `KAM-` too please? 🙏

https://github.com/KammaData/.github/blob/2852d921b49668812b2685deb6507c18f6df699f/.github/pull_request_template.md?plain=1#L12

gemmadlou committed:
KAM-XXXX: change references to the old JIRA prefixes

vicchi reviewed (approved):
Excellent!
